tests/model/test_performance_signature.py

```diff
@@ -0,0 +1,26 @@
+from datetime import datetime
+
+from treeherder.perf.models import PerformanceSignature
+
+
+def test_performance_signatures_with_different_applications(test_perf_signature):
+    assert PerformanceSignature.objects.count() == 1
+
+    # create a performance signature that only differs from another existing one by the application name
+    PerformanceSignature.objects.create(
```

Comment:
This is pretty hard to follow.
How about doing a simple clone, similar to [Django's official recommendations](https://docs.djangoproject.com/en/3.0/topics/db/queries/#copying-model-instances)?
What's left for us is to simply change the application attribute, then save the new instance.